web3/providers/ipc.py

```diff
@@ -91,6 +93,14 @@ def get_default_ipc_path(testnet=False):
         if os.path.exists(ipc_path):
             return ipc_path
 
+        base_trinity_path = Path('~').expanduser() / 'Library' / 'Application Support' / 'trinity'
+        if testnet:
+            ipc_path = base_trinity_path / 'ropsten' / 'jsonrpc.ipc'
```

Comment:
This seems problematic. What happens when trinity adds supports for other networks like `rinkeby`?. A boolean `testnet` won't be sufficient to guess the path.

Reply:
You can, of course, try all possible paths and check if the path exists?
But what happens when a machine is running both `ropsten` and `rinkeby`? :D 
It will always pick the first path it comes across!

Reply:
Yeah, that's problematic.  Wondering if we should do something like:

1. mainnet remains special (and we can also implement some checks to validate that we are indeed connected to the mainnet
2. use a name-based pattern for all the other networks (`rinkeby/kovan/ropsten`) and somehow namespace the imports as `from web3.auto.ropsten import ...`

Reply:
Yup, I am pretty happy with the idea of dropping support for implicitly detecting the testnet IPC (where `from web3.auto.ropsten import w3` is the next-best option for users).